tracer/src/Datadog.Trace/Debugger/Expressions/ProbeProcessor.cs

```diff
@@ -246,14 +267,17 @@ private ExpressionEvaluationResult Evaluate(DebuggerSnapshotCreator snapshotCrea
                 !evaluationResult.HasError &&
                 !ProbeRateLimiter.Instance.Sample(ProbeInfo.ProbeId))
             {
-                snapshotCreator.Stop();
                 shouldStopCapture = true;
                 return evaluationResult;
             }
 
             if (evaluationResult.Metric.HasValue)
             {
-                LiveDebugger.Instance.SendMetrics();
+                LiveDebugger.Instance.SendMetrics(ProbeInfo.MetricKind.Value, ProbeInfo.MetricName, evaluationResult.Metric.Value);
+                // snapshot creator is created for all probes in the method invokers,
+                // if it is a metric probe, once we sent the value, we can stop the invokers working and dispose the snapshot creator
+                snapshotCreator.Dispose();
```

Comment:
I am bit confused, why are you disposing here?

Reply:
Maybe I missing something but as I understand it, once we were got the metric we don't need anymore our invokers till the next execution of course. There is no need of capturing members and sure we don't need any snapshot operation.

Reply:
Fair enough. Resolving.